src/pcm/SoxrResampler.cxx

```diff
@@ -65,6 +85,104 @@ soxr_quality_name(unsigned long recipe) noexcept
 	}
 }
 
+static char *
+quality_advanced_nextparam(char *src, char c)
+{
+        static char *str = NULL;
+        char *ptr, *ret;
+
+        if (src)
+		str = src;
+        if (str && (ptr = strchr(str, c))) {
+                ret = str;
+                *ptr = '\0';
+                str = ptr + 1;
+        } else {
+                ret = str;
+                str = NULL;
+        }
+
+        return ret && ret[0] ? ret : NULL;
+}
+
+static bool
+soxr_parse_quality_advanced(const char *copt, struct soxr *r)  noexcept
+{
+	char *opt, *recipe = NULL, *flags = NULL;
+	char *atten = NULL;
+	char *precision = NULL, *passband_end = NULL, *stopband_begin = NULL, *phase_response = NULL;
+
+	opt = xstrdup(copt);
+
+	r->max_rate = false;
+	r->exception = false;
+
+	if (opt) {
+		recipe = quality_advanced_nextparam(opt, ':');
+		flags = quality_advanced_nextparam(NULL, ':');
+		atten = quality_advanced_nextparam(NULL, ':');
+		precision = quality_advanced_nextparam(NULL, ':');
+		passband_end = quality_advanced_nextparam(NULL, ':');
+		stopband_begin = quality_advanced_nextparam(NULL, ':');
+		phase_response = quality_advanced_nextparam(NULL, ':');
+	}
+
+	/* default to HQ (20 bit) if not user specified */
+	r->q_recipe = SOXR_HQ;
+	r->q_flags = 0;
+	/* default to 1db of attenuation if not user specified */
+	r->scale = pow(10, -1.0 / 20);
+	/* override recipe derived values with user specified values */
+	r->q_precision = 0;
+	r->q_passband_end = 0;
+	r->q_stopband_begin = 0;
+	r->q_phase_response = -1;
+
+	if (recipe && recipe[0] != '\0') {
+		if (strchr(recipe, 'v')) r->q_recipe = SOXR_VHQ;
```

Comment:
So this checks only if a `v` is present, but allows lots of other garbage without complaining. Bad parser.